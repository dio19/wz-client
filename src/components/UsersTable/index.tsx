import React, { useState, useEffect } from "react";
import classNames from "classnames";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TablePagination from '@mui/material/TablePagination';
import Paper from "@material-ui/core/Paper";
import { CircularProgress } from "@material-ui/core";
import { Users, UsersTableProps } from '../../types/interfaces';

const UsersTable = ({isOpen}: UsersTableProps) => {
  const [ users, setUsers ] = useState<Users | undefined>(undefined);
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);

  const getUsers = async () => {
    try {
      const response = await fetch('https://wz-server-dio.herokuapp.com/users')
      const json = await response.json()
      setUsers(json)
    } catch (e) {
      console.error(e);
    }
  }

  useEffect(() => {
    if (users === undefined) {
      getUsers();
    }
  }, [users]);

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  return (
    users === undefined ? <CircularProgress /> : 
    <Paper className={classNames("table-container", {"table-container-collapsed": !isOpen})}>
      <TableContainer>
        <Table stickyHeader aria-label="sticky table">
          <TableHead>
              <TableRow>
                <TableCell align="center">Id</TableCell>
                <TableCell align="center">Name</TableCell>
                <TableCell align="center">Username</TableCell>
                <TableCell align="center">Email</TableCell>
                <TableCell align="center">Phone</TableCell>
              </TableRow>
          </TableHead>
          <TableBody>
            {users && users.data && users.data
            .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
            .map((user) => {
              return (
                <TableRow hover role="checkbox" tabIndex={-1} key={user.id}>
                  <TableCell align="center" component="th" scope="row">
                    {user.id}
                  </TableCell>
                  <TableCell align="center">{user.name}</TableCell>
                  <TableCell align="center">{user.username}</TableCell>
                  <TableCell align="center">{user.email}</TableCell>
                  <TableCell align="center">{user.phone}</TableCell>
                </TableRow>
              );
            })}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={[2, 5, 10]}
        component="div"
        count={users.data.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onPageChange={handleChangePage}
        onRowsPerPageChange={handleChangeRowsPerPage}
      />
    </Paper>
  );
}

export default UsersTable;