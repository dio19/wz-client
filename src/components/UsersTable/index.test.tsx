import { render } from '@testing-library/react';
import UsersTable from './index';

describe('Tests Users Table', () => {
  it('should match snapshot with true value', () => {
    const props = {
      isOpen: true,
    };

    const { asFragment } = render(<UsersTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with false', () => {
    const props = {
      isOpen: false,
    };

    const { asFragment } = render(<UsersTable {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

});