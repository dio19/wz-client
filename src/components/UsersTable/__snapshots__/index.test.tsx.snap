// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Users Table should match snapshot with false 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Tests Users Table should match snapshot with true value 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;
